components/telegram_bot_api/telegram_bot_api.app.js

```diff
@@ -89,5 +271,386 @@ module.exports = {
+     * @param {string} chatId - Unique identifier for the target chat or
+     * username of the target channel (in the format @channelusername)
+     * @param  {String|stream.Stream|Buffer} media - A file path or a Stream. Can
+     * also be a `file_id` previously uploaded
+     * @param {Object} [opts] - An object containing additional configuration
+     * options for this method
+     * @param {Number|String} [opts.filename] - The name of the file to send
+     * @param {Number|String} [opts.contentType] - The MIME type of the file to
+     * send
+     * @param {...*} [opts.extraOpts] - Additional Telegram query options to be
+     * fed to the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api)
+     * @returns {Promise<TelegramBot.Message>} The sent message
+     */
+    async sendMedia(sendFn, chatId, media, opts) {
+      const {
+        filename,
+        contentType,
+        ...extraOpts
+      } = opts;
+      return await sendFn(chatId, media, extraOpts, {
+        filename,
+        contentType,
+      });
+    },
+    /**
+     * @typedef {import("./constants.js").UIMediaType} UIMediaType
+     *
+     */
+    /**
+     * Send a file (Document/Image, Photo, Audio, Video, Video Note, Voice,
+     * Sticker) as the media type specified by the `type` parameter
+     *
+     * @param {UIMediaType} type - The media type of the file
+     * @param {string} chatId - Unique identifier for the target chat or
+     * username of the target channel (in the format @channelusername)
+     * @param {String|stream.Stream|Buffer} media - A file path or a Stream. Can
+     * also be a `file_id` previously uploaded
+     * @param {Object} [opts] - An object containing additional configuration
+     * options for this method
+     * @param {Number|String} [opts.filename] - The name of the file to send
+     * @param {Number|String} [opts.contentType] - The MIME type of the file to
+     * send
+     * @param {...*} [opts.extraOpts] - Additional Telegram query options to be
+     * fed to the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api)
+     * @returns {Promise<TelegramBot.Message>} The sent message
+     */
+    async sendMediaByType(type, chatId, media, opts) {
+      const sdk = this.sdk();
+      const typeToSendFn = {
+        [TELEGRAM_BOT_API_UI_MEDIA_DOCUMENT]: sdk.sendDocument,
+        [TELEGRAM_BOT_API_UI_MEDIA_PHOTO]: sdk.sendPhoto,
+        [TELEGRAM_BOT_API_UI_MEDIA_AUDIO]: sdk.sendAudio,
+        [TELEGRAM_BOT_API_UI_MEDIA_VIDEO]: sdk.sendVideo,
+        [TELEGRAM_BOT_API_UI_MEDIA_VIDEO_NOTE]: sdk.sendVideoNote,
+        [TELEGRAM_BOT_API_UI_MEDIA_VOICE]: sdk.sendVoice,
+        [TELEGRAM_BOT_API_UI_MEDIA_STICKER]: sdk.sendSticker,
+      };
+      const sendFn = typeToSendFn[type].bind(sdk);
+      return this.sendMedia(sendFn, chatId, media, opts);
+    },
+    /**
+     * @type {SendMediaFn}
+     */
```

Comment:
Is this jsdoc really needed? same goes for the rest of functions like this one

Reply:
I added this jsdoc to document the types and descriptions of the parameters and return values. Do you think the code be simpler or more easily understood without them?

Reply:
This is just an opinon. What I would do is to just call the needed function for example right from the action component in this case would be `send-audio-file`
```
this.sendMediaByType(TELEGRAM_BOT_API_UI_MEDIA_AUDIO, chatId, audio, opts);
```
in that way you can remove all those funcion definitions from with only this in the jsdoc
```
    /**
     * @type {SendMediaFn}
     */
```

Reply:
I refactored the [`send<Media>` functions](https://github.com/PipedreamHQ/pipedream/blob/0d608e3513972888420b3f73fcfdcb6fdb8ab067/components/telegram_bot_api/telegram_bot_api.app.mjs#L413) to call `this.sendMedia` directly rather than through `this.sendMediaByType`, and removed the JSDocs. I was hoping to keep separate functions for `sendPhoto`, `sendAudio`, etc., like the [Telegram Bot Node.js module](https://github.com/yagop/node-telegram-bot-api/blob/5cc6629ca02c5160424be4615b281866fb56c027/doc/api.md#TelegramBot+sendPhoto) and [Telegram Bot API](https://core.telegram.org/bots/api#sendphoto) do. But we could remove the separate functions in favor of a `this.sendMediaByType` method if that's preferable.

Reply:
It looks great to me as they are. Thanks for making this change!